examples/tokiomarine2020-a.rs

```diff
@@ -0,0 +1,38 @@
+// https://atcoder.jp/contests/tokiomarine2020/tasks/tokiomarine2020_a
+//
+// 以下のクレートを使用。
+// - `ascii`
+// - `proconio`
+
+use ascii::AsciiString;
+use proconio::input;
+
+fn main() {
+    // Sを[`ascii::AsciiString`]として、[`proconio::input!`]で入力を読む。
+    //
+    // [`ascii::AsciiString`]: https://docs.rs/ascii/1.0.0/ascii/struct.AsciiString.html
+    // [`proconio::input!`]:   https://docs.rs/proconio/0.3.6/proconio/macro.input.html
+    input! {
+        s: AsciiString,
+    }
+
+    // [`AsciiStr`] / `AsciiString`は
+    //
+    // 1. [`usize`の範囲でアクセス可能]であり**かつ**
+    // 2. そのまま[`Display`]可能
+    //
+    // であることから、このような問題で変換を挟まずに簡潔に書くことができる。
+    //
+    // `ascii`クレートを使わずに行うなら、
+    //
+    // 1. Sを`String`として読み、`s.chars().take(3).collect::<String>()`
+    // 2. Sを[`proconio::marker::Bytes`]経由で`Vec<u8>`として読み(あるいは`String`から`.into_bytes()`する)、`std::str::from_utf8(&s[..3]).unwrap()`
```

Comment:
String も `Index<Range<usize>>` 系統を実装しているので (<https://doc.rust-lang.org/std/string/struct.String.html#impl-Index%3CRange%3Cusize%3E%3E>) 、例えば次のようなコードも問題なく実行できるかと思います。

```rust
let s = "hello".to_string();
assert_eq!(&s[..3], "hel");
```

特に理由がなければこれについても触れておくほうが良さそうです (最も簡便と思うので、事情がなければ 1. に配置するほうが良いかも知れません) 。逆に言うとこの問題では `AsciiString` の優位性が見えづらいかもしれませんね...。

```suggestion
    // 1. Sを`String`として読み、`&s[..3]`としてスライスを取得する
    // 2. Sを`String`として読み、`s.chars().take(3).collect::<String>()`
    // 3. Sを[`proconio::marker::Bytes`]経由で`Vec<u8>`として読み(あるいは`String`から`.into_bytes()`する)、`std::str::from_utf8(&s[..3]).unwrap()`
```

Reply:
なるほど... 完全に忘れていました。 `AsciiStr`の優位性を考えてみて`SliceIndex<str>`が行うchar boundaryのチェックをスキップできるというのが思い浮かびましたが、[端だけ見れば良いのでO(1)のようです](https://doc.rust-lang.org/std/slice/trait.SliceIndex.html#implementors)。

本当に存在意義が怪しくなってきましたがとりあえず私からはそのままにしておきます...